### PR TITLE
Scaffold Update - Build on changes

### PR DESCRIPTION
**Pull request by jigypeper**
## Description

Scheduled builds should only build if there are changes on the main branch.
Changes:
- Copier question note to highlight schedule only applies to main
- Build yml updated with changes

Associated risk level (low)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: src/.github/workflows/{% if github_actions and github_actions_build_branches and odoo_oci_image %}build.yml{% endif %}.jinja [hunk 1]
```diff
@@ -21,13 +21,67 @@ on:
     - cron: {{ github_actions_build_on_schedule_cron | tojson }}
   {%- endif %}
 
+{%- if github_actions_build_on_schedule and github_actions_build_on_schedule_cron %}
+permissions:
+  packages: read
```
Comment (theangryangel) [inline src/.github/workflows/{% if github_actions and github_actions_build_branches and odoo_oci_image %}build.yml{% endif %}.jinja]: Is this the correct permission, given we're not using ghcr? Or is this good enough?

File: src/.github/workflows/{% if github_actions and github_actions_build_branches and odoo_oci_image %}build.yml{% endif %}.jinja [hunk 2]
```diff
@@ -21,13 +21,67 @@ on:
...
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set Timestamps
+        id: set-timestamps
+        run: |
+          NOW=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
+          ONE_DAY_AGO=$(date -u -d '-24 hours' +"%Y-%m-%dT%H:%M:%SZ")
+          echo "NOW=$NOW" >> $GITHUB_ENV
+          echo "ONE_DAY_AGO=$ONE_DAY_AGO" >> $GITHUB_ENV
```
Comment (theangryangel) [inline src/.github/workflows/{% if github_actions and github_actions_build_branches and odoo_oci_image %}build.yml{% endif %}.jinja:45]:
```suggestion
          echo "ONE_DAY_AGO=$ONE_DAY_AGO" >> $GITHUB_ENV

```

**Review by theangryangel — DISMISSED** [review]

File: src/.github/workflows/{% if github_actions and github_actions_build_branches and odoo_oci_image %}build.yml{% endif %}.jinja (hunk 1 above)
Comment (jigypeper) [reply src/.github/workflows/{% if github_actions and github_actions_build_branches and odoo_oci_image %}build.yml{% endif %}.jinja]: seems to work on my dummy repo, i think it only needs to read the repo to get the data

Comment (jigypeper) [reply src/.github/workflows/{% if github_actions and github_actions_build_branches and odoo_oci_image %}build.yml{% endif %}.jinja]: wait, scratch that, its wrong

**Pull request merged**